lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -0,0 +1,56 @@
+require "thread"
+
+module CC
+  module Engine
+    module Analyzers
+      class FileThreadPool
+        DEFAULT_CONCURRENCY = 2
+        MAX_CONCURRENCY = 2
+
+        def initialize(files, concurrency: DEFAULT_CONCURRENCY)
+          @files = files
+          @concurrency = concurrency
+        end
+
+        def run(&block)
+          queue = build_queue
+
+          @workers = thread_count.times.map do
+            Thread.new do
+              begin 
+                yield queue.pop(true)
+              rescue ThreadError
```

Comment:
This rescue is actually just for when queue.pop runs out of files to pop, so I don't think it should log to stderr.